addons/website_blog/models/website_blog.py

```diff
@@ -188,29 +154,32 @@ def html_tag_nodes(self, html, attribute=None, tags=None, context=None):
             html = html[5:-6]
         return html, mapping
 
-    def _postproces_content(self, cr, uid, id, content=None, context=None):
+    @api.model
+    def _postprocess_content(self, id, content=None):
+        #(_postproces_content = _postprocess_content)
         if content is None:
-            content = self.browse(cr, uid, id, context=context).content
+            content = self.browse(id).content
         if content is False:
             return content
-
-        content, mapping = self.html_tag_nodes(content, attribute='data-chatter-id', tags=['p'], context=context)
+        content, mapping = self.html_tag_nodes(content, attribute='data-chatter-id', tags=['p'])
         if id:  # not creating
             existing = [x[0] for x in mapping if x[0]]
-            msg_ids = self.pool['mail.message'].search(cr, SUPERUSER_ID, [
+            msg_ids = self.env['mail.message'].sudo().search([
```

Comment:
i think no need of search. directly use self.website_message_ids.filtered(lambda message: message.path not in existing and message.path != False).unlink()